src/javascript/scroll-spy.js

```diff
@@ -9,6 +12,12 @@ export function scrollSpy({ $navSections, currentSectionClass }) {
 
   function setCurrentSection(index) {
     forEachRight($navSections, (navEl, navIndex) => {
+
+      if (navIndex === index &&
+          URIJS(window.location.href).hash() !== jquery(navEl).attr('href')) {
```

Comment:
on teoretycznie może coś więcej ale nie korzystamy z tego więc zaraz to uproszczę.